pkg/ovsdb/transact.go

```diff
@@ -1032,11 +1047,16 @@ func preInsert(txn *Transaction, ovsOp *libovsdb.Operation, ovsResult *libovsdb.
 }
 
 func doInsert(txn *Transaction, ovsOp *libovsdb.Operation, ovsResult *libovsdb.OperationResult) error {
-	for uuid, row := range txn.cache.Table(txn.request.DBName, ovsOp.Table) {
-		if ovsOp.UUIDName != "" && (*row)[COL_UUIDNAME] == ovsOp.UUIDName {
+	if ovsOp.UUIDName != "" {
+		if _, ok := txn.mapUUID[ovsOp.UUIDName]; ok {
 			klog.Errorf("Duplicate uuid-name: %s", ovsOp.UUIDName)
 			return errors.New(E_DUP_UUIDNAME)
 		}
+		uuid := common.GenerateUUID()
+		txn.mapUUID[ovsOp.UUIDName] = uuid
+	}
+
```

Comment:
Where do you handling the option when a user specifies a required UUID for the new row

Reply:
fixed, now handling above